src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchSource.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.GoToBase;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SymbolSearch
+{
+    internal class SymbolSearchSource : ISymbolSourceFromLocation
+    {
+        internal SymbolSearchSourceProvider ServiceProvider { get; }
+
+        public SymbolSearchSource(SymbolSearchSourceProvider symbolSourceProvider)
+        {
+            ServiceProvider = symbolSourceProvider;
+        }
+
+        public ImageId Icon => default;
+
+        public string DescriptionText => string.Empty;
+
+        string INamed.DisplayName => ServicesVSResources.Symbol_search_source_name;
+
+        string ISymbolSource.Id => nameof(SymbolSearchSource);
+
+        public async Task<SymbolSearchStatus> FindSymbolsAsync(string navigationKind, VisualStudio.Language.Intellisense.SymbolSearch.Location sourceLocation, ISymbolSearchCallback callback, CancellationToken token)
+        {
+            // This method is called off the UI thread
+            var snapshot = sourceLocation.PersistentSpan.Document.TextBuffer.CurrentSnapshot;
```

Comment:
How does this chain of .ing work if the document is closed? I'm imagining the TextBuffer is null then?